src/Campaigns/Migrations/MigrateFormsToCampaignForms.php

```diff
@@ -77,17 +77,43 @@ protected function getFormData(): array
         })
             ->whereIsNull('campaigns.id');
 
-        // Exclude forms with an `upgraded` status, which are archived.
-        $query->where('forms.post_status', 'upgraded', '!=');
+        /**
+         * Exclude forms with an `upgraded` status, which are WP revisions.
+         *
+         * @see https://wordpress.org/documentation/article/post-status/#auto-draft
+         */
+        $query->where('forms.post_status', 'auto-draft', '!=');
+
+        /**
+         * Excluded upgraded V2 forms as their corresponding V3 version will be used to create the campaign - later the V2 form will be added to the proper campaign as a non-default form through the addUpgradedV2FormToCampaign() method.
+         */
+        $upgradedFormsIds = $this->getUpgradedV2FormsIds();
```

Comment:
Could this be refactored to a join of the meta with a "where not"? Or do we need a second query?

Reply:
I didn't use join but I refactored it to remove the getUpgradedV2FormsIds method and kept only one query.

Resolved in b1038647d440fcd743c6bde63a4e4a6c96995b46